### PR TITLE
[SPARK-21321][Spark Core] Spark very verbose on shutdown

### PR DESCRIPTION
**Pull request by yoonlee95**
## What changes were proposed in this pull request?

The current code is very verbose on shutdown.

The changes I propose is to change the log level when the driver is shutting down and the RPC connections are closed (RpcEnvStoppedException).

## How was this patch tested?

Tested with word count(deploy-mode = cluster, master = yarn, num-executors = 4) with 300GB of data.

### CODE REVIEW

File: core/src/main/scala/org/apache/spark/rpc/netty/Inbox.scala [hunk 1]
```diff
@@ -206,6 +206,13 @@ private[netty] class Inbox(
       case NonFatal(e) =>
         try endpoint.onError(e) catch {
           case NonFatal(ee) => logError(s"Ignoring error", ee)
+          case NonFatal(ee) =>
+            if (stopped) {
+            logDebug(s"Ignoring error", ee)
```
Comment (srowen) [inline core/src/main/scala/org/apache/spark/rpc/netty/Inbox.scala]: logInfo, not debug, probably. I can see reducing the level if you know your'e stopping, but not in other cases. The code style here would have to be fixed. For example you don't use string interpolation here.

**Review by srowen — COMMENTED** [review]
I still am not sure it's worth revising these. What's logged at shutdown isn't that big.

**Comment by tgravescs** [comment]
There is no reason to print out messages that aren't useful to the users. Many users see Warnings and read them and think there is a problem with their application or configuration.  Most of these are really for advanced users or devs trying to debug what is going on.  So changing them to debug makes sense to me.  

There are also times that shutdown can take a very long time and all its doing it spewing these log messages. For instance when you have a backlog of events in the event queue.  I have seen  multiple minutes of it just throwing the warnings like: Message $message dropped. ${e.getMessage}
dynamic allocation can also add a lot of warning messages that are unneeded.  When I'm shutting down things should stop quickly and release the resources for others to use.

I still need to review in detail but these should just be in the shutdown case, other times I agree that you may want to see these.

We should definitely update the description here and in the jira to have more details though.

**Comment by tgravescs** [comment]
Jenkins, test this please

**Comment by tgravescs** [comment]
@yoonlee95 please look at the test failures, it might be a race in clearing the events

**Comment by tgravescs** [comment]
Jenkins, test this please

**Comment by tgravescs** [comment]
Jenkins, test this please

**Comment by vanzin** [comment]
ok to test

File: core/src/main/scala/org/apache/spark/scheduler/LiveListenerBus.scala [hunk 2]
```diff
@@ -200,7 +200,7 @@ private[spark] class LiveListenerBus(conf: SparkConf) extends SparkListenerBus {
   private def queueIsEmpty: Boolean = synchronized { eventQueue.isEmpty && !processingEvent }
 
   /**
-   * Stop the listener bus. It will wait until the queued events have been processed, but drop the
+   * Stop the listener bus. It will clear the queued events for faster shutdown and drop the
```
Comment (tgravescs) [inline core/src/main/scala/org/apache/spark/scheduler/LiveListenerBus.scala]: remove the comment now

**Comment by tgravescs** [comment]
+1

**Pull request closed without merging**